controllers/helper_test.go

```diff
@@ -538,6 +540,52 @@ func testBuildManagedZone(name, ns, domainName string) *kuadrantdnsv1alpha1.Mana
 	}
 }
 
+// testEndpointsTraversable consumes an array of endpoints and returns a boolean
+// indicating presence of that path from host to all IPs
+func testEndpointsTraversable(endpoints []*endpoint.Endpoint, host string, ips []string) bool {
+	for _, ip := range ips {
+		allTargetsFound := false
+		for _, ep := range endpoints {
+			// the target on the endpoint exists as an endpoint
+			if ep.DNSName == host {
+				// target is our IP
+				if slices.Contains(ep.Targets, ip) {
+					return true
+				} else {
+					// target is not out IP. Use target as a host and check for existence of Endpoints with such a DNSName
+					for _, target := range ep.Targets {
+						// if at least one returns as true allTargetsFound will be locked in true
+						allTargetsFound = allTargetsFound || testEndpointsTraversable(endpoints, target, []string{ip})
+					}
+				}
```

Comment:
Should probably break here

Reply:
We should really check all of the endpoints that match the current host. [Here](https://github.com/Kuadrant/kuadrant-operator/blob/45337827e3b3b41a081ed7639b596381848f3310/controllers/dnspolicy_controller_multi_cluster_test.go#L229-L243), for example, we will fail if we break after the first match if we happened to look for a wrong IP

Reply:
ok, i see